rosette/src/RBLstring.cc

```diff
@@ -153,6 +153,15 @@ Ob* RBLstring::subObject(int start, int n) {
     return result;
 }
 
+bool RBLstring::match_pattern(Ob* other) {
+    PROTECT_THIS(RBLstring);
+    if (!IS_A(other, RBLstring)) {
+        return false;
+    }
+    RBLstring* otherStr = (RBLstring*) other;
+    return strcmp((char*)&SELF->byte(0), (char*)&otherStr->byte(0)) == 0;
```

Comment:
This isn't really a pattern match, is it? Should this method just be called `compare`?

Reply:
Also, please confirm we're using null-terminated strings.

Reply:
We are. It's awful, but changing it is a bit of work.